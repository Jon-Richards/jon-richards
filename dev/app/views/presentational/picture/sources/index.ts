export { Sources, SourcesProps } from './sources';
