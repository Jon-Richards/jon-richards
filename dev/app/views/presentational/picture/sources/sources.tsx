import * as React from 'react';

/**
 * Renders an array of <source> tags.  There are ways to produce this behavior
 * with a single srcset attribute, but this method offers better readability at
 * runtime.
 */
export function Sources(props: SourcesProps): JSX.Element {
  const mapped = props.sources.map(source => (
    <source
      key={source.source}
      media={source.mediaQuery}
      srcSet={source.source}
      data-testid="picture__source"
    />
  ));

  return <React.Fragment>{mapped}</React.Fragment>;
}

export type SourcesProps = {
  sources: Source[];
}

type Source = {
  mediaQuery: string;
  source: string;
}
