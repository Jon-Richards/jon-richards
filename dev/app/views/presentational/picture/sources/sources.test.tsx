import * as React from 'react';
import { Sources, SourcesProps } from './sources';
import { render } from 'enzyme';

describe('The Sources component.', () => {
  it('Should match the snapshot.', () => {
    const sources: SourcesProps['sources'] = [
      { mediaQuery: 'min-width(320px)', source: 'test_small.jpg' },
      { mediaQuery: 'min-width(480px)', source: 'test_medium.jpg' },
      { mediaQuery: 'min-width(720px)', source: 'test_large.jpg' },
    ];
    const test_1 = render(<Sources sources={sources} />);

    expect(test_1).toMatchSnapshot();
  });

  it('Renders a source tag for each provided source.', () => {
    const sources: SourcesProps['sources'] = [
      { mediaQuery: 'min-width(320px)', source: 'test_small.jpg' },
      { mediaQuery: 'min-width(480px)', source: 'test_medium.jpg' },
      { mediaQuery: 'min-width(720px)', source: 'test_large.jpg' },
    ];
    const test_1 = render(<Sources sources={sources} />);

    expect(test_1.length).toBe(3);
  })
});
