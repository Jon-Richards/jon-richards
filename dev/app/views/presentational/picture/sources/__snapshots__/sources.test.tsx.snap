// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Sources component. Should match the snapshot. 1`] = `
Array [
  <source
    data-testid="picture__source"
    media="min-width(320px)"
    srcset="test_small.jpg"
  />,
  <source
    data-testid="picture__source"
    media="min-width(480px)"
    srcset="test_medium.jpg"
  />,
  <source
    data-testid="picture__source"
    media="min-width(720px)"
    srcset="test_large.jpg"
  />,
]
`;
